src/main/java/io/cdap/plugin/gcp/datastore/source/DatastoreRecordReader.java

```diff
@@ -107,4 +114,24 @@ public float getProgress() {
   public void close() {
   }
 
+  // Datastore API only returns up to 300 items. Need to use cursor pagination if there is more results
+  private void loadPage() throws IOException {
+    queryBuilder.setStartCursor(cursor);
+    RunQueryRequest request = RunQueryRequest.newBuilder()
+      .setQuery(queryBuilder)
+      // partition id needs to be set in the RunQueryRequest in addition to being passed to QuerySplitter.getSplits.
+      // This is a quirk of the V1 API.
+      .setPartitionId(partitionIdBuilder)
+      .build();
+    try {
+      QueryResultBatch batch = datastore.runQuery(request).getBatch();
+      batchSizeCounter.increment(batch.getSerializedSize());
+      lastBatchMoreResultsType = batch.getMoreResults();
+      LOG.debug("Batch page moreResultsType: {}", lastBatchMoreResultsType);
```

Comment:
Could you please rephrase, so that looking at the logs one could tell what is this line about without looking at the code? E.g. something along the lines of `Loaded Datastore batch of size {}, more results status: {}`. Also, are we going to get a lot of those (300 seems like a small number). If so, more it to trace, we don't want to spam the logs.